### DISCUSSION TITLE
Call STRAPI API with basic authentication

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7816] [Legacy Topics]
Hello, I would like to call strpi api through username and password, 
1. this username would be added to users collection.
2. Also this username will be added to a role
3. Role will allow permission as needed

This(POSTMAN call) doesnt seem to work for me, and says invalid token, is this feature available in 3.1.1 version or am i doing something wrong?

We can not use JWT token implementation as it involves refreshing of the token everytime.
Your inputs are much appreciated on this.

Kind Regards,
Basu

**Comment by derrickmehaffy** [comment]
Currently we do not support basic authentication, you will first have to login and get the JWT then use that JWT for the requests.

**Reply by basavarajdodamani** [reply]
Thanks @derrickmehaffy for the response. Is there any other way of doing this? we would like restrict users hitting strapi and we can not use public role as it allows for everyone to hit the API.

**Reply by derrickmehaffy** [reply]
It would require you to customize the users-permissions plugin, but there is a ton of code you will have to change as the usage of JWT is used all over the place.

https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#how-do-i-customize-a-plugin

In a case like this, it's not something I would recommend.

**Reply by basavarajdodamani** [reply]
@derrickmehaffy thanks for the response and basic authentication would have been good in our usecase, however i am thinking to use other somewhat relatable feature in this case, is to use IP whitelisting, for which i need to get a go ahead from customer and also shall post if there are any issues with it.

**Reply by derrickmehaffy** [reply]
Yeah certainly, we do have ip whitelisting/blacklisting support so that shouldn't be an issue.
